internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -664,7 +674,31 @@ func (r *AgentServiceConfigReconciler) newAssistedServiceDeployment(instance *ai
 
 		return nil
 	}
-	return deployment, mutateFn
+	return deployment, mutateFn, nil
+}
+
+func getOpenshiftVersions(instance *aiv1beta1.AgentServiceConfig) (string, error) {
+	if instance.Spec.OSImages == nil {
+		return OpenshiftVersions(), nil
+	}
+
+	var versions models.OpenshiftVersions
+	for _, image := range instance.Spec.OSImages {
+		version := models.OpenshiftVersion{
+			RhcosImage: &image.Url,
+			RhcosVersion: &image.Version,
+		}
+		// TODO(djzager) should we be handling the case where multiple Openshift Versions
+		// are specified?
```

Comment:
Not sure what you mean here, why wouldn't we?

Reply:
without doing anything, we'll just overwrite in the case of conflicts. Wasn't clear that we should actively throw an error if we see two "4.8"s

Reply:
Ah sorry I misunderstood "multiple Openshift Versions"
You're talking about a case where someone specifies the same openshift version multiple times.

I was thinking an entry for 4.7 and one for 4.8 would be "multiple Openshift Versions" and it seemed that we were handling that just fine :laughing: